pkg/sdk/deps_builder.go

```diff
@@ -42,12 +46,28 @@ func (sdk *poktrollSDK) buildDeps(
 	}
 	deps = depinject.Configs(deps, depinject.Supply(blockClient))
 
+	urlString := HostToGRPCUrl(config.QueryNodeGRPCUrl)
+
+	var creds credentials.TransportCredentials
+	if strings.HasPrefix(urlString, "grpcs://") {
+		systemRoots, err := x509.SystemCertPool()
+		if err != nil {
+			return nil, err
+		}
+
+		creds = credentials.NewTLS(&tls.Config{
+			RootCAs: systemRoots,
+		})
+	} else {
+		creds = insecure.NewCredentials()
+	}
```

Comment:
`OPTIONAL NIT`: I think we can improve readability by moving this into a small helper and calling

`creds := getTransportCreds()`

Reply:
How to handle the test for this function? It is unexported however the convention in this repo is to put all tests in a separate `sdk_test` package where it wouldn't have access to it.

Reply:
That's fine. The goal of this is to improve code readability and maintainability while reducing cognitive overhead for readers of the source code.

If it makes sense to you, feel free to resolve this thread.